flink-table/flink-sql-gateway/src/test/java/org/apache/flink/table/gateway/service/SqlGatewayServiceITCase.java

```diff
@@ -157,6 +169,80 @@ public void testOpenSessionWithEnvironment() throws Exception {
         assertThat(tableEnv.listModules()).contains(moduleName);
     }
 
+    @Test
+    public void testConfigureSessionWithLegalStatement(@TempDir java.nio.file.Path tmpDir)
+            throws Exception {
+        SessionHandle sessionHandle = service.openSession(defaultSessionEnvironment);
+
+        // SET & RESET
+        verifyConfigureSession(sessionHandle, "SET 'key1' = 'value1';");
+        Map<String, String> config = new HashMap<>();
+        config.put("key1", "value1");
+        assertThat(service.getSessionConfig(sessionHandle)).containsAllEntriesOf(config);
+
+        verifyConfigureSession(sessionHandle, "RESET 'key1';");
+        assertThat(service.getSessionConfig(sessionHandle)).doesNotContainEntry("key1", "value1");
+
+        // CREATE & USE & ALTER & DROP
+        verifyConfigureSession(
+                sessionHandle,
+                "CREATE CATALOG mycat with ('type' = 'generic_in_memory', 'default-database' = 'db');");
+
+        verifyConfigureSession(sessionHandle, "USE CATALOG mycat;");
+        assertThat(service.getCurrentCatalog(sessionHandle)).isEqualTo("mycat");
```

Comment:
Is it worth checking the `type` and/or `default-database`?

Reply:
The `type` is used for discovering the corresponding catalog factory. Here the `generic_in_memory` identify the `GenericInMemoryCatalogFactory`. I think we can verify it by `mycat instance of GenericInMemoryCatalog` but the `SqlGatewayService` doesn't expose the api to get the catalog class but only the name string. Otherwise, creating mycat successfully means the factory has been found, so I think it's not a problem.
Checking `default-database` is added.

Reply:
Makes sense!